analyzer/src/main/scala/com/avsystem/commons/analyzer/ImplicitValueClasses.scala

```diff
@@ -0,0 +1,53 @@
+package com.avsystem.commons
+package analyzer
+
+import scala.tools.nsc.Global
+
+class ImplicitValueClasses(g: Global) extends AnalyzerRule(g, "implicitValueClasses", Level.Warn) {
+
+  import global.*
+  import definitions.*
+
+  private lazy val defaultClasses = Set[Symbol](AnyClass, AnyValClass, ObjectClass)
+
+  private lazy val reportOnNestedClasses = argument match {
+    case null => false
+    case a => a.toBoolean
+  }
```

Comment:
not a fan of the non-descriptive 'true' parameter, some mode enum like `all`, `top-level` would be better (with a default of course)